src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy

```diff
@@ -122,12 +126,21 @@ class NUnit extends ConventionTask {
         project.file(getReportFolder())
     }
 
+    File getLogFolderImpl() {
+        if (logFolder)
+            project.file(logFolder)
+        else
+            return null
+    }
+
     @OutputFile
     File getTestReportPath() {
         // for the non-default nunit tasks, ensure we write the report in a separate file
-        // TODO: Do we need to supply prefix when user has specified its own report file name?
-        def reportFileNamePrefix = name == 'nunit' ? '' : name
```

Comment:
Right, this change isn't needed for this PR, but it solves the problem that i mentioned above. I can open another PR for it if you want.
When making this change I was taking into account the fact that if user didn't specify report file, he doesn't really need it and we can follow the nunit default behavior that store all results in _TestResult.xml_.

There is another solution that keeps the previous behavior. It was written in your _TODO_ comment. Namely, prefix won't be used when user specified his own report file name. I think this is more tolerant solution if you want to keep previous behavior. I will do this if you agree.